code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -260,3 +260,16 @@
 		return 0
 
 	return O.equip(src, visualsOnly)
+
+
+//delete all equipment without dropping anything
+/mob/living/carbon/human/proc/delete_equipment()
+	for(var/slot in get_all_slots())//order matters, dependant slots go first
+		var/obj/item/I = get_item_by_slot(slot)
+		temporarilyRemoveItemFromInventory(I,TRUE)
+		qdel(I)

```

Comment:
item/Destroy calls temporarilyRemoveItemFromInventory itself so this is redundant